src/core/public/chrome/ui/header/collapsible_nav.tsx

```diff
@@ -0,0 +1,262 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiCollapsibleNav,
+  EuiCollapsibleNavGroup,
+  EuiListGroup,
+  EuiListGroupItem,
+  EuiFlexItem,
+  EuiHorizontalRule,
+  EuiShowFor,
+  EuiText,
+} from '@elastic/eui';
+import { groupBy, sortBy } from 'lodash';
+import { OnIsLockedUpdate } from './';
+import { AppCategory } from '../../../../types';
+import { NavLink, RecentNavLink } from './nav_link';
+
+function getAllCategories(allCategorizedLinks: Record<string, NavLink[]>) {
+  const allCategories = {} as Record<string, AppCategory | undefined>;
+
+  for (const [key, value] of Object.entries(allCategorizedLinks)) {
+    allCategories[key] = value[0].category;
+  }
+
+  return allCategories;
+}
+
+function getOrderedCategories(
+  mainCategories: Record<string, NavLink[]>,
+  categoryDictionary: ReturnType<typeof getAllCategories>
+) {
+  return sortBy(
+    Object.keys(mainCategories),
+    categoryName => categoryDictionary[categoryName]?.order
+  );
+}
+
+function getCategoryLocalStorageKey(label: string) {
+  return `core.navGroup.${label}`;
+}
+
+function getIsCategoryOpen(label: string) {
+  const value = localStorage.getItem(getCategoryLocalStorageKey(label)) ?? 'true';
+
+  return value === 'true';
+}
```

Comment:
Why do you think it's better?

The only thing I can think of is that it feels maybe a little neater? But I think the (granted, somewhat minimal) cost of creating/stringifying/parsing objects repeatedly along with the overhead in complexity to read/write isn't worth it. If you've got another reason why though I'm all ears!

Reply:
Main reason would be to keep things in order and avoid 'polluting' the localStorage with a linear number of entries when you can have a constant one instead. Yea, basically just a little neater.

Only other upside I see with having a single key is regarding cleanup: it would be trivial to remove a static key from the storage, where cleaning up in current implementation would need to have the list of categories to do so. (One concrete example would be, in a next version of the nav, some categories where moved/renamed, it would require to keep a reference to the old categories id/label to properly clean the localStorage)

But as I said, this is minor and non blocking. I'm good keeping it that way until core's `localStorage` wrapper do this work under the hood

Reply:
Going to mark this issue resolved then. Will migrate to core's wrapper when available 👍